qtp_target_gene/validate.py

```diff
@@ -22,8 +24,48 @@
     'FASTA_Sanger': ({'raw_fasta'}, set()),
 }
 
+MUST_GZ = [
+    # raw input files: FASTQ, per_sample_FASTQ
+    'raw_forward_seqs', 'raw_barcodes', 'raw_reverse_seqs', 'raw_fasta',
+    # preprocessed files: demultiplexed, trimmed
+    'preprocessed_fastq', 'preprocessed_fasta']
 
-def _validate_multiple(qclient, job_id, prep_info, files, atype):
+
+def _gzip_file(filepath, test=False):
+    """gzip the given filepath if needed
+
+    Parameters
+    ----------
+    filepath : string
+        The filepath to verify or compress
+    test : bolean
+        If True do not compress but change the filename
+
+    Returns
+    -------
+    str
+        the new gz filepath, None if error
+    str
+        the error, None if success
+    """
+    error = None
+    return_fp = None
+    if test:
+        return_fp = '%s.gz' % filepath
+    else:
+        if not is_zipfile(filepath):
+            gz_cmd = 'gzip -f %s' % filepath
```

Comment:
Should we be using pigz or are we not guaranteed to have access to multiple cores here?

Reply:
Not guaranteed and in fact all validate jobs run single core.